aws-lc-rs/src/ec/key_pair.rs

```diff
@@ -305,42 +285,30 @@ impl Debug for PrivateKey<'_> {
     }
 }
 
-pub struct EcPrivateKeyBinType {
-    _priv: (),
-}
-/// Elliptic curve private key data encoded as a big-endian fixed-length integer.
-pub type EcPrivateKeyBin = Buffer<'static, EcPrivateKeyBinType>;
-
-pub struct EcPrivateKeyRfc5915DerType {
-    _priv: (),
-}
-/// Elliptic curve private key as a DER-encoded `ECPrivateKey` (RFC 5915) structure.
-pub type EcPrivateKeyRfc5915Der = Buffer<'static, EcPrivateKeyRfc5915DerType>;
-
-impl AsBigEndian<EcPrivateKeyBin> for PrivateKey<'_> {
+impl AsBigEndian<EcPrivateKeyBin<'static>> for PrivateKey<'_> {
     /// Exposes the private key encoded as a big-endian fixed-length integer.
     ///
     /// For most use-cases, `EcdsaKeyPair::to_pkcs8()` should be preferred.
     ///
     /// # Errors
     /// `error::Unspecified` if serialization failed.
-    fn as_be_bytes(&self) -> Result<EcPrivateKeyBin, Unspecified> {
+    fn as_be_bytes(&self) -> Result<EcPrivateKeyBin<'static>, Unspecified> {
         unsafe {
             let buffer = ec::marshal_private_key_to_buffer(
-                self.0.algorithm.id,
+                self.0.algorithm.id.private_key_size(),
                 &self.0.evp_pkey.as_const(),
             )?;
             Ok(EcPrivateKeyBin::new(buffer))
         }
     }
 }
 
-impl AsDer<EcPrivateKeyRfc5915Der> for PrivateKey<'_> {
+impl AsDer<EcPrivateKeyRfc5915Der<'static>> for PrivateKey<'_> {
```

Comment:
What does adding `'static` do?

Reply:
In this case, `'static` essentially means that the value "owns" all of the data it references. Specifically, the [`Cow`](https://doc.rust-lang.org/std/borrow/enum.Cow.html) value that the `Buffer` (thinly) wraps must be a [`Cow::Owned`](https://doc.rust-lang.org/std/borrow/enum.Cow.html#variant.Owned)